### PR TITLE
Add the Containerfile

### PR DESCRIPTION
**Pull request by brunoapimentel**
Only a minimal configuration was set to the Containerfile, we will need to add more dependencies as other package managers are onboarded.

# Maintainers will complete the following section

- [ ] Commit messages are descriptive enough
- [n/a] Code coverage from testing does not decrease and new code is covered
- [n/a] New code has type annotations
- [ ] Docs updated (if applicable)

### CODE REVIEW

File: .github/workflows/gating.yaml
Comment (brunoapimentel) [inline .github/workflows/gating.yaml:98]: Do we want to keep ignoring this rule, or do we want to pin the RPM versions in the Containerfile?

File: Containerfile
Comment (brunoapimentel) [inline Containerfile:5]: I removed every RPM I did not know why it was there :D

Although I verified that the built cachi2 image is working in a simple scenario, please double check if I didn't remove anything important for the `fetch-deps --package gomod` feature.

File: Containerfile
Comment (chmeliik) [inline Containerfile:18]: Doesn't the `COPY . .` already copy everything including .git? Should we have a containerignore that excludes things, or copy only the things we need?

File: Containerfile
Comment (chmeliik) [inline Containerfile:1]: Should we try `fedora-minimal:36`?

https://fedoramagazine.org/build-smaller-containers/

File: .github/workflows/gating.yaml
Comment (chmeliik) [reply .github/workflows/gating.yaml:98]: Hmm, I feel like pinning the RPMs without something like dependabot to update them has more downsides than benefits.

We'll need a process to check for CVEs and manually update the versions of affected RPMs. When bumping Fedora versions, we'll need to change all versions

File: Containerfile
Comment (chmeliik) [reply Containerfile:5]: Looks correct to me :+1:

File: Containerfile
Comment (chmeliik) [reply Containerfile:17]: Oh! I didn't notice we had one already!

Could we add `.mypy_cache/` to the ignore file? Out of the missing ignores, that seems to be the biggest one, 15.3 MiB on my laptop

+:100: for mounting the .git directory

Comment (chmeliik) [reply Containerfile:17]: Does it work for you? I get a setuptools-scm error :confused:

```
  LookupError: setuptools-scm was unable to detect version for /src.
  
  Make sure you're either building from a fully intact git repository or PyPI tarballs. Most other sources (such as GitHub's tarballs, a git checkout without the .git folder) don't contain the necessary metadata and will not work.
```

Comment (chmeliik) [reply Containerfile:17]: I think it's a SELinux permissions things, adding `Z` worked (the same `Z` used with `--volume=src:dest:Z`
```
RUN --mount=type=bind,source=.git,target=/src/.git,Z \
```

Comment (brunoapimentel) [reply Containerfile:17]: So, as we did in https://github.com/containerbuildsystem/cachito/pull/742, I have removed the mount in favor of copying the .git folder to the image.

File: .github/workflows/gating.yaml
Comment (taylormadore) [inline .github/workflows/gating.yaml:92]: If https://github.com/containerbuildsystem/cachi2/pull/5 gets merged first, we might want to update to actions/checkout@v3

File: Containerfile
Comment (taylormadore) [inline Containerfile:17]: Should we rm -rf the .git directory after the cachi2 package is installed?

**Comment by taylormadore** [comment]
Do we also need to update the README to point to this new image in quay or is that covered in a different task?

File: .github/workflows/gating.yaml
Comment (MartinBasti) [reply .github/workflows/gating.yaml:92]: I'd go with v3 even before merging #5

**Comment by brunoapimentel** [comment]
> Do we also need to update the README to point to this new image in quay or is that covered in a different task?

I would leave it to @chmeliik's [PR](https://github.com/containerbuildsystem/cachi2/pull/8).

**Review by chmeliik — APPROVED** [review]
LGTM :+1:

**Comment by taylormadore** [comment]
LGTM

**Review by taylormadore — APPROVED** [review]

**Pull request merged**
